### DISCUSSION TITLE
Warnings Encountered During Operation of DataFactoryTools

### DISCUSSION THREAD

**Discussion by EkueAF** [Azure-Player/azure.datafactory.tools#415] [Q&A] [answered]
Dear @NowinskiK ,

I trust this message finds you well. :-) 

We have successfully implemented your tool, DataFactoryTools, and it has been wonderful to use. 

However, we have encountered some warnings during operation which seem to be related to updates and naming issues in commands. I was wondering if you are aware of these warnings and if there are plans to address them in future updates?

Here are the warnings we encountered:

1. Both Az and AzureRM modules were detected on this machine. Az and AzureRM modules cannot be imported in the same session or used in the same script or runbook.
2. You're using AzureRM which will be retired soon, please schedule an update.
3. The names of some imported commands from the module Microsoft.Azure.PowerShell.Cmdlets.Network include unapproved verbs that might make them less discoverable.
4. The names of some imported commands from the module Az.Network include unapproved verbs that might make them less discoverable.
5. You're using Az.DataFactory version 1.18.4. The latest version of Az.DataFactory is 1.18.6. Upgrade your Az modules using the following commands.
6. Upcoming breaking changes in the cmdlet Get-AzAccessToken.

Looking forward to your guidance on these matters.

Best regards,

Ekué

**Comment by NowinskiK** [comment, marked as answer]
Thanks for this question.
All warnings (except no.5) come from your environment where you run #adftools - probably Azure DevOps Agent.
Since ver.2.0 (DevOps tasks):
```
Related Az.* modules must be installed on DevOps Agent (no longer belongs to tasks)
```
Link: https://github.com/Azure-Player/azure.datafactory.devops#version-2

Therefore, you must ensure you have an appropriate version of the required modules, or remove old ones.

**Reply by morrisNEB** [reply]
Hi, I'm receiving these same warnings when running the adftools publish task. In my case I'm using an Azure hosted agent to run the task and I specify 'windows-latest' for the agent version. It's not clear how to upgrade or remove powershell modules on a hosted agent.

**Reply by NowinskiK** [reply]
Why? It's very well documented - for example:
[Uninstall the AzureRM module](https://learn.microsoft.com/en-us/powershell/azure/uninstall-az-ps?view=azps-12.3.0#uninstall-the-azurerm-module)
